watcher/diffWatcher.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// DiffWatcher regularly check data at path provided, notifies if changes detected.
+type DiffWatcher struct {
```

Comment:
The naming of that file is a bit odd for Go? I think I normally see files named like `diff_watcher.go`, say?

But I'd tweak a few other things related to naming... This exported struct would be referred to as `watcher.DiffWatcher`, which has some stuttering? Could be `watcher.Diff`, maybe? And then this file could be just `diff.go`, and that would make sense?

Reply:
Renamed.